### PR TITLE
test: add mock social connector

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Split yijun's original branch `xiaoyijun-log-3831-add-mock-social-connector`
Add mock social connector to unblock rest integration tests @IceHe 

## Testing

N/A mock connector not referenced yet
@logto-io/eng

### CODE REVIEW

**Review by IceHe — APPROVED** [review]
LGTM

**Review by darcyYe — APPROVED** [review]

File: packages/connector-mock-social/src/constant.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    en: 'Social mock connector description',
+    'zh-CN': 'Mock 社交登录连接器的描述',
+    'ko-KR': 'Social mock connector description',
+    'tr-TR': 'Social mock connector description',
+  },
+  readme: './README.md',
+  configTemplate: './docs/config-template.json',
+};
+
+export const defaultTimeout = 5000;
```
Comment (IceHe) [inline packages/connector-mock-social/src/constant.ts:26]: unused constant

File: packages/connector-mock-social/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  public validateConfig(config: unknown): asserts config is MockSocialConfig {
+    const result = mockSocialConfigGuard.safeParse(config);
+
+    if (!result.success) {
+      throw new ConnectorError(ConnectorErrorCodes.InvalidConfig, result.error);
+    }
+  }
+
+  public getAuthorizationUri: GetAuthorizationUri = async ({ state, redirectUri }) => {
+    return `http://mock.social.com?state=${state}&redirect_uri=${redirectUri}`;
```
Comment (wangsijie) [inline packages/connector-mock-social/src/index.ts:44]:
```suggestion
    return `http://mock.social.com/?state=${state}&redirect_uri=${redirectUri}`;
```

**Review by wangsijie — APPROVED** [review]

File: packages/connector-mock-social/src/constant.ts (hunk 1 above)
Comment (simeng-li) [reply packages/connector-mock-social/src/constant.ts]: updated

File: packages/connector-mock-social/src/index.ts (hunk 2 above)
Comment (simeng-li) [reply packages/connector-mock-social/src/index.ts]: updated

**Pull request merged**